nimble/host/services/audio/pacs/src/ble_audio_svc_pacs.c

```diff
@@ -0,0 +1,434 @@
+
+    buf = os_mbuf_extend(ctxt->om, 4);
+    if (buf == NULL) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    put_le32(buf + 0, ble_svc_audio_pacs_sink_audio_locations);
+
+    return 0;
+}
+
+static int
+ble_svc_audio_pacs_source_audio_loc_read_access(struct ble_gatt_access_ctxt *
+                                                ctxt)
+{
+    uint8_t *buf;
+
+    buf = os_mbuf_extend(ctxt->om, 4);
+    if (buf == NULL) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    put_le32(buf + 0, ble_svc_audio_pacs_source_audio_locations);
+
+    return 0;
+}
+
+static struct available_ctx *
+ble_svc_audio_pacs_find_avail_ctx(uint16_t conn_handle)
+{
+    int i;
+
+    for (i = 0; i < MYNEWT_VAL(BLE_MAX_CONNECTIONS); i++) {
+        if (ble_svc_audio_pacs_avail_contexts[i].conn_handle == conn_handle) {
+            return &ble_svc_audio_pacs_avail_contexts[i];
+        }
+    }
+    return NULL;
+}
+
+static int
+ble_svc_audio_pacs_avail_audio_ctx_read_access(uint16_t conn_handle,
+                                               struct ble_gatt_access_ctxt *ctxt)
+{
+    struct available_ctx *avail_ctx;
+    uint8_t *buf;
+
+    avail_ctx = ble_svc_audio_pacs_find_avail_ctx(conn_handle);
+
+    buf = os_mbuf_extend(ctxt->om, 4);
+    if (buf == NULL) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    put_le16(buf + 0, avail_ctx->ble_svc_audio_pacs_avail_sink_contexts);
+    put_le16(buf + 2, avail_ctx->ble_svc_audio_pacs_avail_source_contexts);
+
+    return 0;
+}
+
+static int
+ble_svc_audio_pacs_sup_audio_ctx_read_access(struct ble_gatt_access_ctxt
+                                             *ctxt)
+{
+    uint8_t *buf;
+
+    buf = os_mbuf_extend(ctxt->om, 4);
+    if (buf == NULL) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    put_le16(buf + 0, ble_svc_audio_pacs_sup_sink_contexts);
+    put_le16(buf + 2, ble_svc_audio_pacs_sup_source_contexts);
+
+    return 0;
+}
+
+static int
+ble_svc_audio_pacs_access(uint16_t conn_handle, uint16_t attr_handle,
+                          struct ble_gatt_access_ctxt *ctxt, void *arg)
+{
+    uint16_t uuid16 = ble_uuid_u16(ctxt->chr->uuid);
+    int rc;
+
+    switch (uuid16) {
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_PAC:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_sink_pac_read_access(ctxt);
+        } else {
+            assert(0);
+            rc = BLE_ATT_ERR_UNLIKELY;
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_AUDIO_LOCATIONS:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_sink_audio_loc_read_access(ctxt);
+        } else {
+            rc = BLE_ATT_ERR_REQ_NOT_SUPPORTED;
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SOURCE_PAC:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_source_pac_read_access(ctxt);
+        } else {
+            assert(0);
```

Comment:
Is it GATT which should protect this code from getting other prop than READ or this is up to remote device?

If remote device is able to crash us, it is not good.

Reply:
add err (unlikely, not permitted etc)